### PR TITLE
api/permissions: Allow user to GET their own permissions

### PR DESCRIPTION
**Pull request by ssangervasi**
Required for #1480 

### Code Changes

* This change updates the GET USER_PERMISSIONS endpoint to only require the USER_PERMISSION_READ scope when requesting the permissions for _another_ user. If the User Id of the client credentials matches the User Id of the request, the scope check is skipped.
* Cleaned up tests for the endpoint and added case for current-user case
    * The test functions were inconsistent about creating the user and permissions object that needed to be queried.
    * Some were using a `url` fixture that was an invalid string - it contained `{user_id}` instead of actually passing the id.

### Steps to Confirm

* [ ] `GET /api/v1/user/{user_id}/permission` when authenticated as the root user should return the users's permissions.
* [ ] When authenticated as a user with limited permissions, this should only succeed with the current user Id and raise a `403` for any other.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

Issue #1480 highlights that we never re-validate the auth token saved in localStorage. Requests that require auth return `403` if the token gets invalidated on the backend, but this isn't enough information for the frontend to tell whether the _token was invalid_ or if it just doesn't have permission for that one scope.

The User Permission endpoint seems like a robust solution: if a user can't get their own permissions, then they must not have a valid token. We could also use this in the future to toggle features that only make sense if the user has, for example, `PRIVACY_REQUEST_DELETE` permissions.

<details>
<summary>Alternative solutions/implementations</summary>

1. Log out on any endpoint `403` (or `401`)
   - I thought through this option, but it just creates more problems. It's perfectly reasonable for a user to have access to _some_ endpoints block without kicking them out of the app.
   - Also, some endpoints authenticate accounts other than Fides itself. For example, AWS scanning returns 403 if the credentials failed. So a global handler wouldn't work, resulting in a piecemeal solution.
2. Use the Login endpoint - Nope, that's a POST with credentials.
3. Use the GET USER_DETAIL endpoint
   - This was a strong contender, and we might want to allow the user to query themselves anyway.
   - However, for auth checking, the user's permissions seem like the most important information.
   - Also, the USER_DETAIL endpoint is implemented in fideslib, so it's a bit more annoying to change.
</details>

### CODE REVIEW

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (TheAndrewJackson) [inline tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:307]: I might be mistaken but I think the response object may already a helper that does this. Either a property or a method.

```suggestion
        response_body = response.json()
```

```python
response_body = response.json
```

If it doesn't then ignore this.

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (TheAndrewJackson) [inline tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:287]: It looks like the `FidesUser.create` block is used 4 times to make the `auth_user`. This could potentially become a fixture that handles creating and deleting the `auth_user` for each test

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (TheAndrewJackson) [inline tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:196]: Maybe this fixture could clear up the user after each test.

I think it could be something like this

```python
    @pytest.fixture(scope="function")
    def user(self, db) -> Generator:
        user = FidesUser.create(
            db=db,
            data={"username": "user_1", "password": "test_password"},
        )
        yield user
        user.delete(db)
```

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (TheAndrewJackson) [inline tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:202]: This could also clean up the after once the test is finished

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (ssangervasi) [reply tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:229]: I was wondering that. Does our test suite not clean up tables between tests? This looked like a redundant line to me, because I don't see a lot of manual deletions of records in our tests.

Comment (TheAndrewJackson) [reply tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:229]: AFAIK it doesn't. I could be wrong though. There are other tests in the test suite that are clearing out records that are created during the test. Like this one for example

https://github.com/ethyca/fides/blob/a7a5f971289bd98d2c6248e9c7950d8d18c085b3/tests/ops/api/v1/endpoints/test_email_endpoints.py#L96-L112

My guess is that some tests do it and some don't. It generally won't cause issues but it's a good thing to do to keep tests reliable.

Comment (ssangervasi) [reply tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:229]: Sound like it [does](https://ethyca.slack.com/archives/CBKK8TS74/p1666740056522809?thread_ts=1666732882.446949&cid=CBKK8TS74). Maybe those lines are left over from before the cleanup was implemented? That's pretty batteries-included with most test suites though.

**Comment by ssangervasi** [comment]
> For some reason I'm getting a 500 as a response with no traceback in the server logs. Were you able to call the route from postman?

Yeah, I did before I wrote the tests. Are you including a valid auth header? I noticed that it 500's instead of nicely handling a jank token, even before this change.

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (TheAndrewJackson) [reply tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:229]: That's great. It looks like not everyone, myself included, is aware that it data is automatically cleaned up after each test. 

I looked into the `clear_db_tables` and it uses the `autouse` param. That makes it so all tests use that fixture even if they don't request it. [docs example](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request)

Maybe we should have a tech debt ticket for cleaning up tests that are manually deleting data? It would be good to have everything be consistent.

**Comment by TheAndrewJackson** [comment]
@ssangervasi I think that's it. I tried copying the token from the browser. Manually creating a token in postman worked.

**Review by TheAndrewJackson — APPROVED** [review]
LGTM! 🎉 I like the test refactors!

**Comment by TheAndrewJackson** [comment]
I notice the pylint lint failing after I approved it. It should be good to merge once that is fixed.

**Comment by TheAndrewJackson** [comment]
@ssangervasi It looks like a fix was just merged to handle malformed tokens properly #1523

File: tests/ops/api/v1/endpoints/test_user_permission_endpoints.py
Comment (ThomasLaPiana) [reply tests/ops/api/v1/endpoints/test_user_permission_endpoints.py:229]: @TheAndrewJackson for sure open a ticket! We can add this to the testing/quality epic for this quarter

**Pull request merged**
